CMakeLists.txt

```diff
@@ -1,332 +1,374 @@
+# Provides:
+#   (1) finds Git package
+#   (2) Sets ${NGEN_HAS_GIT_DIR} to ON/OFF if ${NGEN_ROOT_DIR}/.git exists
+#   (3) Function `git_update_submodule` which updates a given submodule
+include(GitUpdateSubmodules)
+
+# Options =====================================================================
+include(CMakeDependentOption)
+
+option(NGEN_WITH_MPI         "Build with MPI support"             OFF)
+option(NGEN_WITH_NETCDF      "Build with NetCDF support"          ON)
+option(NGEN_WITH_SQLITE      "Build with SQLite3 support"         OFF)
+option(NGEN_WITH_UDUNITS     "Build with UDUNITS2 support"        ON)
+option(NGEN_WITH_BMI_FORTRAN "Build with Fortran BMI support"     OFF)
+option(NGEN_WITH_BMI_C       "Build with C BMI support"           ON)
+option(NGEN_WITH_PYTHON      "Build with embedded Python support" OFF)
+option(NGEN_WITH_TESTS       "Build with unit tests"              ON)
+option(NGEN_QUIET            "Silence output"                     OFF)
+
+# These options require dependency on some of the above options
+# Syntax: cmake_dependent_option(<option> "<help_text>" <value> <depends> <force>)
+cmake_dependent_option(NGEN_WITH_ROUTING "Build with t-route integration" ON "NGEN_WITH_PYTHON" OFF)
+cmake_dependent_option(NGEN_UPDATE_GIT_SUBMODULES "Update submodules on configure" ON "GIT_FOUND;NGEN_HAS_GIT_DIR" OFF)
+
+option(BMI_FORTRAN_ISO_C_LIB_DIR "Directory hint for middleware Fortran shared lib handling iso_c_binding" "${NGEN_ROOT_DIR}/extern/iso_c_fortran_bmi/cmake_build")
+option(BMI_FORTRAN_ISO_C_LIB_NAME "Name for middleware Fortran shared lib handling iso_c_binding" "iso_c_bmi")
+
+include(DeprecatedOptions)
+deprecated_option(OLD MPI_ACTIVE            NEW NGEN_WITH_MPI     INCLUDE_ENV)
+deprecated_option(OLD UDUNITS_ACTIVE        NEW NGEN_WITH_UDUNITS INCLUDE_ENV)
+deprecated_option(OLD NGEN_ACTIVATE_PYTHON  NEW NGEN_WITH_PYTHON  INCLUDE_ENV)
+deprecated_option(OLD NGEN_ACTIVATE_ROUTING NEW NGEN_WITH_ROUTING)
+deprecated_option(OLD NETCDF_ACTIVE         NEW NGEN_WITH_NETCDF)
+deprecated_option(OLD BMI_C_LIB_ACTIVE      NEW NGEN_WITH_BMI_C)
+deprecated_option(OLD BMI_FORTRAN_ACTIVE    NEW NGEN_WITH_BMI_FORTRAN)
+deprecated_option(OLD QUIET                 NEW NGEN_QUIET)
+deprecated_option(OLD PACKAGE_TESTS         NEW NGEN_WITH_TESTS)
+# Project =====================================================================
+
+# -----------------------------------------------------------------------------
+# Set global C++ options
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+
+# Define project version and use via generated config header
 project(ngen VERSION 0.1.0)
-configure_file(include/NGenConfig.h.in include/NGenConfig.h)
 
-if("${PROJECT_NAME}" STREQUAL "${CMAKE_PROJECT_NAME}")
-    option(PACKAGE_TESTS "Build automated tests" ON)
+configure_file("${NGEN_INC_DIR}/NGenConfig.h.in" "${NGEN_INC_DIR}/NGenConfig.h")
+
+add_executable(ngen "${NGEN_SRC_DIR}/NGen.cpp")
```

Comment:
It would vaguely make sense to move this and everything that touches the target `ngen` down to `src/CMakeLists.txt`, but I don't see that as essential